src/app/clusters/scenes/SceneTable.h

```diff
@@ -0,0 +1,279 @@
+                  TransitionTime100ms time100ms = 0) :
+            mSceneTransitionTime(time),
+            mTransitionTime100(time100ms)
+        {
+            this->SetName(sceneName);
+            mExtentsionFieldsSets = fields;
+        }
+        SceneData(const SceneData & other) :
+            mSceneTransitionTime(other.mSceneTransitionTime), mTransitionTime100(other.mTransitionTime100)
+        {
+            this->SetName(other.mNameSpan);
+            mExtentsionFieldsSets = other.mExtentsionFieldsSets;
+        }
+        ~SceneData(){};
+
+        CHIP_ERROR Serialize(TLV::TLVWriter & writer) const
+        {
+            TLV::TLVType container;
+            ReturnErrorOnFailure(writer.StartContainer(TLV::ContextTag(1), TLV::kTLVType_Structure, container));
+
+            // A 0 size means the name wasn't used so it won't get stored
+            if (!this->mNameSpan.empty())
+            {
+                ReturnErrorOnFailure(writer.PutString(TagSceneName(), this->mNameSpan));
+            }
+
+            ReturnErrorOnFailure(writer.Put(TagSceneTransitionTime(), static_cast<uint16_t>(this->mSceneTransitionTime)));
+            ReturnErrorOnFailure(writer.Put(TagSceneTransitionTime100(), static_cast<uint8_t>(this->mTransitionTime100)));
+            ReturnErrorOnFailure(this->mExtentsionFieldsSets.Serialize(writer));
+
+            return writer.EndContainer(container);
+        }
+        CHIP_ERROR Deserialize(TLV::TLVReader & reader)
+        {
+            TLV::TLVType container;
+            ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::ContextTag(1)));
+            ReturnErrorOnFailure(reader.EnterContainer(container));
+
+            ReturnErrorOnFailure(reader.Next());
+            TLV::Tag currTag = reader.GetTag();
+            VerifyOrReturnError(TagSceneName() == currTag || TagSceneTransitionTime() == currTag, CHIP_ERROR_WRONG_TLV_TYPE);
+
+            // If there was no error, a name is expected from the storage, if there was an unexpectec TLV element,
+            if (currTag == TagSceneName())
+            {
+                ReturnErrorOnFailure(reader.Get(this->mNameSpan));
+                this->SetName(this->mNameSpan);
+                ReturnErrorOnFailure(reader.Next(TagSceneTransitionTime()));
+            }
+
+            ReturnErrorOnFailure(reader.Get(this->mSceneTransitionTime));
+            ReturnErrorOnFailure(reader.Next(TagSceneTransitionTime100()));
+            ReturnErrorOnFailure(reader.Get(this->mTransitionTime100));
+            ReturnErrorOnFailure(this->mExtentsionFieldsSets.Deserialize(reader));
+
+            return reader.ExitContainer(container);
+        }
+
+        void SetName(const CharSpan & sceneName)
+        {
+            if (nullptr == sceneName.data())
+            {
+                mName[0]    = 0;
+                mNameLength = 0;
+            }
+            else
+            {
+                Platform::CopyString(mName, sceneName);

```

Comment:
This will null-terminate. PLease be careful

Reply:
Fixed in latest commit